iree/samples/dynamic_shapes/main.c

```diff
@@ -0,0 +1,290 @@
+        iree_runtime_call_outputs_pop_front_buffer_view(&call, out_buffer_view);
+  }
+
+  iree_runtime_call_deinitialize(&call);
+  return status;
+}
+
+iree_status_t run_sample(iree_string_view_t bytecode_module_path,
+                         iree_string_view_t driver_name) {
+  iree_status_t status = iree_ok_status();
+
+  //===-------------------------------------------------------------------===//
+  // Instance configuration (this should be shared across sessions).
+  fprintf(stdout, "Configuring IREE runtime instance and '%s' device\n",
+          driver_name.data);
+  iree_runtime_instance_options_t instance_options;
+  iree_runtime_instance_options_initialize(IREE_API_VERSION_LATEST,
+                                           &instance_options);
+  iree_runtime_instance_options_use_all_available_drivers(&instance_options);
+  iree_runtime_instance_t* instance = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_instance_create(&instance_options,
+                                          iree_allocator_system(), &instance);
+  }
+  // TODO(#5724): move device selection into the compiled modules.
+  iree_hal_device_t* device = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_instance_try_create_default_device(
+        instance, driver_name, &device);
+  }
+  //===-------------------------------------------------------------------===//
+
+  //===-------------------------------------------------------------------===//
+  // Session configuration (one per loaded module to hold module state).
+  fprintf(stdout, "Creating IREE runtime session\n");
+  iree_runtime_session_options_t session_options;
+  iree_runtime_session_options_initialize(&session_options);
+  iree_runtime_session_t* session = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_session_create_with_device(
+        instance, &session_options, device,
+        iree_runtime_instance_host_allocator(instance), &session);
+  }
+  iree_hal_device_release(device);
+
+  fprintf(stdout, "Loading bytecode module at '%s'\n",
+          bytecode_module_path.data);
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_session_append_bytecode_module_from_file(
+        session, bytecode_module_path.data);
+  }
+  //===-------------------------------------------------------------------===//
+
+  //===-------------------------------------------------------------------===//
+  // Call the exported sample functions with some test inputs
+  fprintf(stdout, "Calling functions\n\n");
+
+  // reduce_sum(1, 10, 100])
+  if (iree_status_is_ok(status)) {
+    const int input[3] = {1, 10, 100};
+    int result = -1;
+    reduce_sum(session, input, 3, &result);
+    fprintf(stdout, "reduce_sum([1, 10, 100]): %d\n", result);
+  }
+
+  // iota(10)
+  if (iree_status_is_ok(status)) {
+    iree_hal_buffer_view_t* result_buffer_view = NULL;
+    iota(session, 10, &result_buffer_view);
+
+    // Map the result buffer and print its elements.
```

Comment:
I think it'd be better to keep these simple and closer to the python (which just prints the results) - for someone getting started the buffer fprint is probably where they would start, and learning how to access the data can be a dedicated sample (talking about mapping, what to do when the returned buffers are not mappable, etc). It's just noise here (50 extra lines of code)